docs/yaml/objects/cfg_data.yaml

```diff
@@ -41,8 +41,15 @@ methods:
       type: str
       description: The name of the variable to set
     value:
-      type: bool
-      description: The value to set as either `1` or `0`
+      type: bool | int
+      description: |
+        The value to set as either `1` or `0`
+
+        Since 0.61.0 passing a number is deprecated, use only booleans or the
+        `.set` method instead.
+
+        *NOTE:* numeric values < 0 have the surprising behavior of being
+        converted to `true`.
```

Comment:
It is not deprecated "since 0.61". And, the `NOTE:` can be described in yaml as

```yaml
notes:
  - numeric values < 0 have [...]
```

But better yet we can make it a `warnings:`

Actually I'd like the entire thing to be a warnings:

```
warnings:
  - Historically, meson would accidentally allow passing a number instead of a boolean. This has the
    surprising behavior of converting to `true` for both numbers < 0 and numbers > 1. It was never
    supposed to be accepted and should never be used. Instead, use only booleans or the
    [[cfg_data.set]] method.
```